src/allmydata/test/test_storage.py

```diff
@@ -1302,56 +1318,58 @@ def secrets(n):
-              {0: ([], [(200, "make me bigger")], None)}, [])
+        write(b"si1", secrets(0),
+              {0: ([], [(200, b"make me bigger")], None)}, [])
         self.compare_leases_without_timestamps(all_leases, list(s0.get_leases()))
 
-        write("si1", secrets(0),
-              {0: ([], [(500, "make me really bigger")], None)}, [])
+        write(b"si1", secrets(0),
+              {0: ([], [(500, b"make me really bigger")], None)}, [])
         self.compare_leases_without_timestamps(all_leases, list(s0.get_leases()))
 
     def test_remove(self):
         ss = self.create("test_remove")
-        self.allocate(ss, "si1", "we1", self._lease_secret.next(),
+        self.allocate(ss, b"si1", b"we1", next(self._lease_secret),
                       set([0,1,2]), 100)
         readv = ss.remote_slot_readv
         writev = ss.remote_slot_testv_and_readv_and_writev
-        secrets = ( self.write_enabler("we1"),
-                    self.renew_secret("we1"),
-                    self.cancel_secret("we1") )
+        secrets = ( self.write_enabler(b"we1"),
+                    self.renew_secret(b"we1"),
+                    self.cancel_secret(b"we1") )
         # delete sh0 by setting its size to zero
-        answer = writev("si1", secrets,
+        answer = writev(b"si1", secrets,
                         {0: ([], [], 0)},
                         [])
         # the answer should mention all the shares that existed before the
         # write
         self.failUnlessEqual(answer, (True, {0:[],1:[],2:[]}) )
         # but a new read should show only sh1 and sh2
-        self.failUnlessEqual(readv("si1", [], [(0,10)]),
-                             {1: [""], 2: [""]})
+        self.failUnlessEqual(readv(b"si1", [], [(0,10)]),
+                             {1: [b""], 2: [b""]})
 
         # delete sh1 by setting its size to zero
-        answer = writev("si1", secrets,
+        answer = writev(b"si1", secrets,
                         {1: ([], [], 0)},
                         [])
         self.failUnlessEqual(answer, (True, {1:[],2:[]}) )
-        self.failUnlessEqual(readv("si1", [], [(0,10)]),
-                             {2: [""]})
+        self.failUnlessEqual(readv(b"si1", [], [(0,10)]),
+                             {2: [b""]})
 
         # delete sh2 by setting its size to zero
-        answer = writev("si1", secrets,
+        answer = writev(b"si1", secrets,
                         {2: ([], [], 0)},
                         [])
         self.failUnlessEqual(answer, (True, {2:[]}) )
-        self.failUnlessEqual(readv("si1", [], [(0,10)]),
+        self.failUnlessEqual(readv(b"si1", [], [(0,10)]),
                              {})
         # and the bucket directory should now be gone
-        si = base32.b2a("si1")
+        si = base32.b2a(b"si1")
         # note: this is a detail of the storage server implementation, and
         # may change in the future
+        if PY3:
+            si = si.decode("utf-8")
```

Comment:
Does a use of `bytes_to_native_string` make sense here?

Reply:
Good idea.